// @openai/agents-realtime/src/index.ts
import * as utilImport from '@openai/agents-realtime/src/utils';

export { RealtimeAgent, RealtimeAgentConfiguration } from '@openai/agents-realtime/src/realtimeAgent';

export {
  RealtimeSession,
  RealtimeSessionConnectOptions,
  RealtimeSessionOptions,
  RealtimeContextData,
} from '@openai/agents-realtime/src/realtimeSession';

export { RealtimeSessionEventTypes } from '@openai/agents-realtime/src/realtimeSessionEvents';

export {
  TransportEvent,
  TransportLayerAudio,
  TransportLayerResponseCompleted,
  TransportLayerResponseStarted,
  TransportLayerTranscriptDelta,
  TransportError,
  TransportToolCallEvent,
  RealtimeTransportEventTypes,
} from '@openai/agents-realtime/src/transportLayerEvents';

export {
  RealtimeClientMessage,
  RealtimeAudioFormat,
  RealtimeSessionConfig,
} from '@openai/agents-realtime/src/clientMessages';

export {
  OpenAIRealtimeWebRTC,
  OpenAIRealtimeWebRTCOptions,
  WebRTCState,
} from '@openai/agents-realtime/src/openaiRealtimeWebRtc';

export {
  OpenAIRealtimeWebSocket,
  OpenAIRealtimeWebSocketOptions,
  WebSocketState,
} from '@openai/agents-realtime/src/openaiRealtimeWebsocket';

export {
  OpenAIRealtimeModels,
  OpenAIRealtimeBase,
  OpenAIRealtimeBaseOptions,
  OpenAIRealtimeEventTypes,
  DEFAULT_OPENAI_REALTIME_MODEL,
  DEFAULT_OPENAI_REALTIME_SESSION_CONFIG,
} from '@openai/agents-realtime/src/openaiRealtimeBase';

export { RealtimeOutputGuardrail } from '@openai/agents-realtime/src/guardrail';

export {
  RealtimeItem,
  RealtimeToolCallItem,
  RealtimeMessageItem,
  RealtimeBaseItem,
} from '@openai/agents-realtime/src/items';

export {
  ApiKey,
  RealtimeTransportLayerConnectOptions,
  RealtimeTransportLayer,
} from '@openai/agents-realtime/src/transportLayer';

export const utils = {
  base64ToArrayBuffer: utilImport.base64ToArrayBuffer,
  arrayBufferToBase64: utilImport.arrayBufferToBase64,
  getLastTextFromAudioOutputMessage:
    utilImport.getLastTextFromAudioOutputMessage,
};

// Re-exporting some core functionalities requires for building front-end
// realtime agents
export {
  FunctionTool,
  ModelBehaviorError,
  OutputGuardrailTripwireTriggered,
  tool,
  UserError,
} from '@openai/agents-core';
