// @openai/agents-core/src/index.ts
import { addTraceProcessor } from '@openai/agents-core/src/tracing';
import { defaultProcessor } from '@openai/agents-core/src/tracing/processor';

export { RuntimeEventEmitter } from '@openai/agents-core/_shims';
export {
  Agent,
  AgentConfiguration,
  AgentConfigWithHandoffs,
  AgentOptions,
  AgentOutputType,
  ToolsToFinalOutputResult,
  ToolToFinalOutputFunction,
  ToolUseBehavior,
  ToolUseBehaviorFlags,
} from '@openai/agents-core/src/agent';
export { Computer } from '@openai/agents-core/src/computer';
export {
  AgentsError,
  GuardrailExecutionError,
  InputGuardrailTripwireTriggered,
  MaxTurnsExceededError,
  ModelBehaviorError,
  OutputGuardrailTripwireTriggered,
  ToolCallError,
  UserError,
  SystemError,
} from '@openai/agents-core/src/errors';
export {
  RunAgentUpdatedStreamEvent,
  RunRawModelStreamEvent,
  RunItemStreamEvent,
  RunStreamEvent,
} from '@openai/agents-core/src/events';
export {
  defineOutputGuardrail,
  GuardrailFunctionOutput,
  InputGuardrail,
  InputGuardrailFunction,
  InputGuardrailFunctionArgs,
  InputGuardrailMetadata,
  InputGuardrailResult,
  OutputGuardrail,
  OutputGuardrailDefinition,
  OutputGuardrailFunction,
  OutputGuardrailFunctionArgs,
  OutputGuardrailMetadata,
  OutputGuardrailResult,
} from '@openai/agents-core/src/guardrail';
export {
  getHandoff,
  getTransferMessage,
  Handoff,
  handoff,
  HandoffInputData,
} from '@openai/agents-core/src/handoff';
export { assistant, system, user } from '@openai/agents-core/src/helpers/message';
export {
  extractAllTextOutput,
  RunHandoffCallItem,
  RunHandoffOutputItem,
  RunItem,
  RunMessageOutputItem,
  RunReasoningItem,
  RunToolApprovalItem,
  RunToolCallItem,
  RunToolCallOutputItem,
} from '@openai/agents-core/src/items';
export { AgentHooks } from '@openai/agents-core/src/lifecycle';
export { getLogger } from '@openai/agents-core/src/logger';
export {
  getAllMcpTools,
  invalidateServerToolsCache,
  MCPServer,
  MCPServerStdio,
  MCPServerStreamableHttp,
} from '@openai/agents-core/src/mcp';
export {
  MCPToolFilterCallable,
  MCPToolFilterContext,
  MCPToolFilterStatic,
  createMCPToolStaticFilter,
} from '@openai/agents-core/src/mcpUtil';
export {
  Model,
  ModelProvider,
  ModelRequest,
  ModelResponse,
  ModelSettings,
  ModelSettingsToolChoice,
  SerializedHandoff,
  SerializedTool,
  SerializedOutputType,
} from '@openai/agents-core/src/model';
export { setDefaultModelProvider } from '@openai/agents-core/src/providers';
export { RunResult, StreamedRunResult } from '@openai/agents-core/src/result';
export {
  IndividualRunOptions,
  NonStreamRunOptions,
  run,
  RunConfig,
  Runner,
  StreamRunOptions,
} from '@openai/agents-core/src/run';
export { RunContext } from '@openai/agents-core/src/runContext';
export { RunState } from '@openai/agents-core/src/runState';
export {
  HostedTool,
  ComputerTool,
  computerTool,
  HostedMCPTool,
  hostedMcpTool,
  FunctionTool,
  FunctionToolResult,
  Tool,
  tool,
  ToolExecuteArgument,
} from '@openai/agents-core/src/tool';
export * from '@openai/agents-core/src/tracing';
export { getGlobalTraceProvider, TraceProvider } from '@openai/agents-core/src/tracing/provider';
/* only export the types not the parsers */
export type {
  AgentInputItem,
  AgentOutputItem,
  AssistantMessageItem,
  HostedToolCallItem,
  ComputerCallResultItem,
  ComputerUseCallItem,
  FunctionCallItem,
  FunctionCallResultItem,
  JsonSchemaDefinition,
  ReasoningItem,
  ResponseStreamEvent,
  SystemMessageItem,
  TextOutput,
  UnknownContext,
  UnknownItem,
  UserMessageItem,
  StreamEvent,
  StreamEventTextStream,
  StreamEventResponseCompleted,
  StreamEventResponseStarted,
  StreamEventGenericItem,
} from '@openai/agents-core/src/types';
export { Usage } from '@openai/agents-core/src/usage';

/**
 * Exporting the whole protocol as an object here. This contains both the types
 * and the zod schemas for parsing the protocol.
 */
export * as protocol from '@openai/agents-core/src/types/protocol';

/**
 * Add the default processor, which exports traces and spans to the backend in batches. You can
 * change the default behavior by either:
 * 1. calling addTraceProcessor, which adds additional processors, or
 * 2. calling setTraceProcessors, which sets the processors and discards the default one
 */
addTraceProcessor(defaultProcessor());
